CMakeLists.txt

```
@@ -487,6 +487,51 @@ if (CODECOVERAGE)
   add_link_options(--coverage)
 endif (CODECOVERAGE)
 
+# TAP test support
+option(TAP "Enable TAP test support" ON)
+
+if (TAP)
+  include(FindPerlLibs)
+  find_package(PerlLibs 5.8)
+
+  if (PERLLIBS_FOUND)
+    get_filename_component(PERL_BIN_PATH ${PERL_EXECUTABLE} DIRECTORY)
+
+    find_program(PROVE prove
+      PATHS
+      ${PERL_BIN_PATH}
+      "/usr/bin"
+      NO_DEFAULT_PATH)
+
+    if (NOT PROVE)
+      message(FATAL_ERROR "The prove binary was not found. Disable TAP tests with -DTAP=OFF.")
+    endif ()
+    #execute_process(COMMAND ${PERL_EXECUTABLE} -MIPC::Run -e ""
+    #  ERROR_QUIET RESULT_VARIABLE PERL_MODULE_STATUS)
+
+    find_file(PERL_IPC_RUN Run.pm
+      PATHS
+      "${PERL_PRIVLIB}"
+      "${PERL_UPDATE_PRIVLIB}"
+      "${PERL_VENDORLIB}"
+      "${PERL_SITELIB}"
+      "/usr/share/perl"
+      "/usr/share/perl5"
+      "/usr/share/perl6"
+      "$ENV{HOME}/perl/lib/perl"
+      "$ENV{HOME}/perl5/lib/perl5"
+      "$ENV{HOME}/perl6/lib/perl6"
+      PATH_SUFFIXES
+      IPC)
+
+    if (NOT PERL_IPC_RUN)
+      message(FATAL_ERROR "IPC::Run Perl module not found. Disable TAP tests with -DTAP=OFF.")
+    endif ()
```

If there is a file `Run.pm` in the top directories, it will find that first. Using `PATH_SUFFIXES` just provides *extra* directories to search. I would suggest that you use the method you commented out instead to check that the library is installed:
```suggestion
execute_process(
    COMMAND perl -mIPC::Run -e ''
    RESULT_VARIABLE status
    ERROR_QUIET)
    # If status is non-zero (true), module does not exist.
    if (status)
      message(FATAL_ERROR "IPC::Run Perl module not found. Disable TAP tests with -DTAP=OFF.")
    endif ()
```